core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -406,7 +406,7 @@ private[spark] class Executor(
 
           // Don't forcibly exit unless the exception was inherently fatal, to avoid
           // stopping other tasks unnecessarily.
-          if (Utils.isFatalError(t)) {
+          if (Utils.isFatalError(t) && !Utils.isLinkageError(t)) {
```

Comment:
It's conceivable this could just be 'whitelisted' as a non-fatal error, but that bears checking where else this is used. I'm not sure I agree with the premise of the change though.